### ISSUE TITLE
build cloudimage and  edge_coreimage fail

### ISSUE THREAD

**Issue by dreamflown** [kubeedge/kubeedge#310]
**Which jobs are failing**:
when I trying to build a image of edgecontroller, there is a error that the package cannot be found,  which shows 
```
can't load package: package github.com/kubeedge/kubeedge/cloud/edgecontroller/cmd: cannot find package "github.com/kubeedge/kubeedge/cloud/edgecontroller/cmd" in any of:
	/usr/local/go/src/github.com/kubeedge/kubeedge/cloud/edgecontroller/cmd (from $GOROOT)
	/go/src/github.com/kubeedge/kubeedge/cloud/edgecontroller/cmd (from $GOPATH)
```
**Which test(s) are failing**:

**Since when has it been failing**:

*Reason for failure**:

**Anything else we need to know**:
And the GOPATH is /usr/local/go/src and the GOROOT is /usr/local/go.
BTW, the project path is /usr/local/go/src/github.com/kubeedge/kubeedge.
The Dockerfile is in kubeedge/build/cloud.
HELP please!

**Comment by tedli** [comment]
@dreamflown

**Comment by dreamflown** [comment]
Do you mean that I should run the comman "docker build" in the kubeedge`s root directory (/github.com/kubeedge/kubeedge) or copy the Dockerfile into the directory "/github.com/kubeedge/kubeedge" then build the image?
thanks for your help.

**Comment by dreamflown** [comment]
> @dreamflown
> Is this error produced by `make cloudimage` in kubeedge's root directory? Or you just use the docker file in somewhere else (other directories)
> 
> Because of there is a `COPY` line
> 
> ```dockerfile
> COPY . /go/src/github.com/kubeedge/kubeedge
> ```
> 
> in the docker file, which would copy the kubeedge's root directory into a container then build the binary inside the container. If we use the docker file to build the image not in the kubeedge's root directory the `COPY .` won't copy the project structure into the container correctly, so `go build` will complain that can't load package.

**Comment by dreamflown** [comment]
> @dreamflown
> Is this error produced by `make cloudimage` in kubeedge's root directory? Or you just use the docker file in somewhere else (other directories)
> 
> Because of there is a `COPY` line
> 
> ```dockerfile
> COPY . /go/src/github.com/kubeedge/kubeedge
> ```
> 
> in the docker file, which would copy the kubeedge's root directory into a container then build the binary inside the container. If we use the docker file to build the image not in the kubeedge's root directory the `COPY .` won't copy the project structure into the container correctly, so `go build` will complain that can't load package.

Yes, it produced by the procedure when I trying to `make cloudimage` , but the command is `docker build ` instead of `make', right？

**Comment by tedli** [comment]
@dreamflown 

There are rules to make cloud image and edge image in `Makefile`, which expressive enough to show how to build these images by using `docker build`.

As alternative, is free to build the image without `make`, `docker build` works, and not need to copy any Dockerfiles to project's root directory.

We can just specify docker build `context` by docker's command args, as `--help` shows:

```
docker build --help

Usage:	docker build [OPTIONS] PATH | URL | -
......
......
TL;DR
```

So the command below will work as you expect:

```
docker build -f <path-to-kubeedge-root/build/<cloud|edge>/Dockerfile> <path-to-kubeedge-root>
```

**Comment by sids-b** [comment]
@dreamflown , I think your GOPATH should be /usr/local/go instead of /usr/local/go/src .

**Comment by dreamflown** [comment]
> @dreamflown , I think your GOPATH should be /usr/local/go instead of /usr/local/go/src .

  @sids-b Actually，it came to my mind the first moment , but it didn`t work.
